sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessorReceiveActions.cs

```diff
@@ -155,6 +171,14 @@ internal async Task CancelMessageLockRenewalAsync()
             }
         }
 
+        private CancellationTokenSource TrackAndArmCancellationSourceIfNecessary(ServiceBusReceivedMessage message)
+        {
+            var messageLockLostCancellationSource = new CancellationTokenSource();
```

Comment:
would it be worthwhile trying to avoid creating cancellation token sources for messages that do not expire?

Reply:
It may?   What messages would not expire?  Not sure that I'm following the scenario.

Reply:
I was wondering if there are scenarios where the client might track messages that are not under peek lock. In such a scenario `LockUntil` would be set to `default` or am I mistaken? In such a scenario, this code path would potentially create unnecessary token sources. But I definitely have to give this another look.

Reply:
I did check once more and it seems for all messages received via the receive actions (except the peeked ones) they are tracked through `TrackMessagesAsReceived`. Therefore we always should have a cancellation token source for those.

For messages that are received under `ReceiveAndDelete` the delivery is disposed and marked as `settled = true`. But interestingly, there is nothing in the receive actions that actually checks for that. So it seems we are still spawning renewal attempts even though the message is already been settled.

Reply:
> For messages that are received under ReceiveAndDelete the delivery is disposed and marked as settled = true.

We don't autorenew lock if the receive mode is not PeekLock - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs#L265